flutter_native_log_handler/android/src/main/kotlin/de/goddchen/flutter/flutter_native_logs/FlutterNativeLogsPlugin.kt

```diff
@@ -33,14 +34,18 @@ class FlutterNativeLogsPlugin : FlutterPlugin, MethodCallHandler, StreamHandler
             Runtime.getRuntime().exec("logcat -c")
             logcatProcess = Runtime.getRuntime().exec("logcat")
             logcatProcess?.let {
-                it
-                    .inputStream
-                    .bufferedReader()
-                    .useLines { lines ->
-                        lines.forEach { line ->
-                            Handler(Looper.getMainLooper()).post { eventSink?.success(line) }
+                try {
+                    it
+                        .inputStream
+                        .bufferedReader()
+                        .useLines { lines ->
+                            lines.forEach { line ->
+                                Handler(Looper.getMainLooper()).post { eventSink?.success(line) }
+                            }
                         }
-                    }
+                } catch (e: IOException) {
+                    // Don't send logs to the channel
```

Comment:
question: does it make sense here to close the stream as well?

Reply:
This seems to make sense so that there are no memory leaks